src/pg4n/config_parser.py

```diff
@@ -0,0 +1,72 @@
+import re
+from typing import Optional, TextIO
+
+from .config_values import ConfigValues
+
+
+class ConfigParser:
+    def __init__(self, file: TextIO):
+        self.file: TextIO = file
+
+    def parse(self) -> Optional[ConfigValues]:
+        """
+        Reads config values from file givein in __init__.
+        """
+
+        optnames = [x.lower() for x in ConfigValues.__annotations__.keys()]
+        config_values: ConfigValues = {}
+
+        option_matcher = re.compile(
+            r"[ \t]*(?P<optname>\w+)[ \t]+(?P<optval>1|0|true|false|yes|no)[ \t]*$",
+            flags=re.IGNORECASE,
+        )
+
+        # Needed for bytes containing files
+        self.file.seek(0)
+
+        for line in self.file.readlines():
+            # To make this work with bytes objects like TemporaryFile contents
+            if isinstance(line, bytes):
+                line = bytes.decode(line, "utf-8")
+            line = str(line)
+            if match := option_matcher.match(line):
```

Comment:
Nipotusta: jos tehdään em. muutos.
```suggestion
            if match := _option_matcher.match(line):
```